README.md

```diff
@@ -1,37 +1,206 @@
+    - :money_with_wings: **Wifi and metered network detection** to ensure the model updates do not use all the available data quota 
+    - :no_bell: All of these smart defaults are easily are **overridable**
+- :mortar_board: Support for both reactive and callback patterns so you have your freedom of choice (_in progress_)
+- :lock: Support for **secure multi-party computation** and **secure aggregation** protocols using **peer-to-peer WebRTC** connections (_in progress_).
 
-## Set-up
+There are a variety of additional privacy-preserving protections that may be applied, including [differential privacy](https://towardsdatascience.com/understanding-differential-privacy-85ce191e198a), [muliti-party computation](https://www.inpher.io/technology/what-is-secure-multiparty-computation), and [secure aggregation](https://research.google/pubs/pub45808/).
 
-You can work on the project by running `pod install` in the root directory. Then open the file `SwiftSyft.xcworkspace` in Xcode. When the project is open on Xcode, you can work on the `SwiftSyft` pod itself in `Pods/Development Pods/SwiftSyft/Classes/*`
+[OpenMined](https://openmined.org) set out to build the **world's first open-source ecosystem for federated learning on web and mobile**. SwiftSyft is a part of this ecosystem, responsible for bringing secure federated learning to iOS devices. You may also train models on Android devices using [KotlinSyft](https://github.com/OpenMined/KotlinSyft) or in web browsers using [syft.js](https://github.com/OpenMined/syft.js).
+
+If you want to know how scalable federated systems are built, [Towards Federated Learning at Scale](https://arxiv.org/pdf/1902.01046.pdf) is a fantastic introduction!
+
+## Installation
+We have not currently made our initial release. SwiftSyft would soon be available on Cocoapods.
+
+## Quick Start
+As a developer, there are few steps to building your own secure federated learning system upon the OpenMined infrastructure:
+
+1. :robot: Generate your secure ML model using [PySyft](https://github.com/OpenMined/PySyft). By design, PySyft is built upon PyTorch and TensorFlow so you **don't need to learn a new ML framework**. You will also need to write a training plan (training code the worker runs) and an averaging plan (code that PyGrid runs to average the model diff).
+2. :earth_americas: Host your model and plans on [PyGrid](https://github.com/OpenMined/PyGrid) which will deal with all the federated learning components of your pipeline. You will need to set up a PyGrid server somewhere, please see their installation instructions on how to do this.
+3. :tada: Start training on the device!
+
+**:notebook: The entire workflow and process is described in greater detail in our [project roadmap](https://github.com/OpenMined/Roadmap/blob/master/web_and_mobile_team/projects/federated_learning.md).**
+
+You can use SwiftSyft as a front-end or as a background service. The following is a quick start example usage:
+
+```swift
```

Comment:
I love all this. But can we make the spacing on these lines 2 spaces instead of 4? That might be default for your editor, but it makes me scroll horizontally to read the code example.